File: Password Factory/Controls/PasswordTextView.m
//
//  PasswordTextView.m
//  Password Factory
//
//  Created by Cristiana Yambo on 11/30/17.
//  Copyright © 2017 Cristiana Yambo. All rights reserved.
//

#import "PasswordTextView.h"
#import "Utilities.h"
@implementation PasswordTextView

-(void)setText:(NSString *)text {
    //remove all newlines because they don't make sense in the password field
    text = [text stringByReplacingOccurrencesOfString:@"\n" withString:@""];
    text = [text stringByReplacingOccurrencesOfString:@"\r" withString:@""];
    //color the text if necessary
    NSAttributedString *s = [Utilities colorText:text size:self.textSize];
    [self.textStorage setAttributedString:s];
}
@end

File: Password Factory iOS/Controls/PasswordTextView.swift
//
//  PasswordTextView.swift
//  Password Factory iOS
//
//  Created by Cristiana Yambo on 12/4/17.
//  Copyright © 2017 Cristiana Yambo. All rights reserved.
//

import UIKit
@IBDesignable

/// TextView for the password display
class PasswordTextView: UITextView {
    public override func layoutSubviews() {
        super.layoutSubviews()
        setDisplay()
    }
    override func prepareForInterfaceBuilder() {
        setDisplay()
    }
    func setDisplay() {
        Utilities.roundCorners(layer: self.layer, withBorder: true)
    }
}
